pallets/messages/src/lib.rs

```diff
@@ -344,28 +351,39 @@ impl<T: Config> Pallet<T> {
 		schema_id: SchemaId,
 		current_block: BlockNumberFor<T>,
 	) -> Result<bool, DispatchError> {
-		<Messages<T>>::try_mutate(
-			current_block,
-			schema_id,
-			|existing_messages| -> Result<bool, DispatchError> {
-				// first message for any schema_id is going to trigger an event
-				let need_event = existing_messages.len() == 0;
-				let index = BlockMessageIndex::<T>::get();
-				let msg = Message {
-					payload, // size is checked on top of extrinsic
-					provider_msa_id,
-					msa_id,
-					index,
-				};
+		let mut metadata = MessageBlockMetadata::<T>::get();
+		ensure!(
+			u32::from(metadata.total_index) < T::MaxMessagesPerBlock::get(),
+			Error::<T>::TooManyMessagesInBlock
+		);
+		let mut found = false;
+		let mut current_index: MessageIndex = 0;
+		for schema_count in metadata.schema_counts.iter_mut() {
+			if schema_count.schema_id == schema_id {
```

Comment:
Nit: You can use `find`
metadata.schema_counts.iter_mut().find(.....

Reply:
No applicable anymore